### PR TITLE
init documentation for `cancelAnimationFrame()` and `requestAnimationFrame()` of `DedicatedWorkerGlobalScope` interface

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

see https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame and https://developer.mozilla.org/en-US/docs/Web/API/Window/cancelAnimationFrame

reference to https://html.spec.whatwg.org/multipage/imagebitmap-and-animations.html#animation-frames

using some content from `Window.requestAnimationFrame()` and `Window.cancelAnimationFrame()` and the example from the specification

### Additional details

Relates to #1519

### CODE REVIEW

File: files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    ctx = transferredCanvas.getContext("2d");
+    self.requestAnimationFrame(draw);
+  }
+  if (e.data.type === "stop") {
+    self.cancelAnimationFrame(handle);
+  }
+});
+```
+
+Finally, when necessary, we could send a message of "stop" to the worker to stop the animation:
```
Comment (teoli2003) [inline files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md:78]:
```suggestion
Finally, if needed, the main thread can send a `"stop"`message to the worker to stop the animation:
```

File: files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- {{domxref("DedicatedWorkerGlobalScope.requestAnimationFrame()")}}
+- {{domxref("Window.requestAnimationFrame()")}}
```
Comment (teoli2003) [inline files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md:97]: This one is not needed.
```suggestion
```

File: files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
+---
+title: "DedicatedWorkerGlobalScope: requestAnimationFrame() method"
+short-title: requestAnimationFrame()
+slug: Web/API/DedicatedWorkerGlobalScope/requestAnimationFrame
+page-type: web-api-instance-method
+browser-compat: api.DedicatedWorkerGlobalScope.requestAnimationFrame
+---
+
```
Comment (teoli2003) [inline files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md:8]: Most comments from `cancelRequestAnimationFrame` also apply here. I'll make a second pass once they are fixed in both pages.

File: files/en-us/web/api/window/cancelanimationframe/index.md [hunk 4]
```diff
@@ -67,4 +67,6 @@ cancelAnimationFrame(myReq);
 
 ## See also
 
-- {{domxref("window.requestAnimationFrame()")}}
+- {{domxref("Window.requestAnimationFrame()")}}
+- {{domxref("DedicatedWorkerGlobalScope.cancelAnimationFrame()")}}
+- {{domxref("DedicatedWorkerGlobalScope.requestAnimationFrame()")}}
```
Comment (teoli2003) [inline files/en-us/web/api/window/cancelanimationframe/index.md:72]:
```suggestion
```

File: files/en-us/web/api/window/requestanimationframe/index.md [hunk 5]
```diff
@@ -172,8 +172,8 @@ function animate() {
 ## See also
 
 - {{domxref("Window.cancelAnimationFrame()")}}
-- {{domxref("OffscreenCanvas")}}
-- [DedicatedWorkerGlobalScope](/en-US/docs/Web/API/DedicatedWorkerGlobalScope)
+- {{domxref("DedicatedWorkerGlobalScope.cancelAnimationFrame()")}}
```
Comment (teoli2003) [inline files/en-us/web/api/window/requestanimationframe/index.md:175]:
```suggestion
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
I have a few questions here, and I'll need to do a second pass once all is fixed, but this is close to be mergeable.

File: files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  - : The ID value returned by the call to {{domxref("DedicatedWorkerGlobalScope.requestAnimationFrame()", "requestAnimationFrame()")}} that requested the callback.
+
+### Return value
+
+None ({{jsxref("undefined")}}).
+
+### Exceptions
+
+- `NotSupportedError` {{domxref("DOMException")}}
+  - : Thrown if the method is not supported in current worker.
```
[the remark this answers is not in the document]
Comment (skyclouds2001) [reply files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md]: and this expection is not required to documented in window. and window. , as in window scope, the exception will never throw

File: files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+## Syntax
+
+```js-nolint
+cancelAnimationFrame(handle)
+```
+
+### Parameters
+
+- `handle`
+  - : The ID value returned by the call to {{domxref("DedicatedWorkerGlobalScope.requestAnimationFrame()", "requestAnimationFrame()")}} that requested the callback.
```
[the remark this answers is not in the document]
Comment (skyclouds2001) [reply files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md:24]: yes, the handle must be created in the same dedicated worker

File: files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+slug: Web/API/DedicatedWorkerGlobalScope/cancelAnimationFrame
+page-type: web-api-instance-method
+browser-compat: api.DedicatedWorkerGlobalScope.cancelAnimationFrame
+---
+
+{{APIRef}}
+
+The **`cancelAnimationFrame()`** method of the {{domxref("DedicatedWorkerGlobalScope")}} interface cancels an animation frame request previously scheduled through a call to {{domxref("DedicatedWorkerGlobalScope.requestAnimationFrame()", "requestAnimationFrame()")}}.
+
+Calling `cancelAnimationFrame()` method requires that current worker need to support this method, which means that current worker is created by {{domxref("Window", "window")}} or the creator of current worker (the worker must be a dedicated worker) also supports this method.
```
Comment (skyclouds2001) [inline files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md:13]:
```suggestion
Calling `cancelAnimationFrame()` method requires that the current worker need to support this method, which means that the current worker is created by {{domxref("Window", "window")}} or by a dedicated worker that also supports this method.
```

File: files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+The frequency of calls to the callback function will generally match the display refresh rate. The most common refresh rate is 60hz, (60 cycles/frames per second), though 75hz, 120hz, and 144hz are also widely used. `requestAnimationFrame()` calls are paused in most browsers when running in background tabs or hidden {{HTMLElement("iframe")}}s, in order to improve performance and battery life.
+
+Your callback function must call `requestAnimationFrame()` again if you want to animate another frame. `requestAnimationFrame()` is one-shot.
+
+> **Warning:** Be sure always to use the first argument (or some other method for getting the current time) to calculate how much the animation will progress in a frame — **otherwise, the animation will run faster on high refresh-rate screens**. For ways to do that, see the examples below.
+
+Calling `requestAnimationFrame()` method requires that current worker need to support this method, which means that current worker is created by {{domxref("Window", "window")}} or the creator of current worker (the worker must be a dedicated worker) also supports this method.
```
Comment (skyclouds2001) [inline files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md:19]:
```suggestion
Calling `requestAnimationFrame()` method requires that the current worker need to support this method, which means that the current worker is created by {{domxref("Window", "window")}} or by a dedicated worker that also supports this method.
```

File: files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+slug: Web/API/DedicatedWorkerGlobalScope/cancelAnimationFrame
+page-type: web-api-instance-method
+browser-compat: api.DedicatedWorkerGlobalScope.cancelAnimationFrame
+---
+
+{{APIRef}}
+
+The **`cancelAnimationFrame()`** method of the {{domxref("DedicatedWorkerGlobalScope")}} interface cancels an animation frame request previously scheduled through a call to {{domxref("DedicatedWorkerGlobalScope.requestAnimationFrame()", "requestAnimationFrame()")}}.
+
+Calling `cancelAnimationFrame()` method requires that the current worker need to support this method, which means that the current worker is created by {{domxref("Window", "window")}} or by a dedicated worker that also supports this method.
```
Comment (skyclouds2001) [inline files/en-us/web/api/dedicatedworkerglobalscope/cancelanimationframe/index.md:13]:
```suggestion
Calling `cancelAnimationFrame()` method requires that the current worker need to has an associated owner {{domxref("Window", "window")}}, which means that the current worker must be created by {{domxref("Window", "window")}} or by a dedicated worker that also has an associated owner {{domxref("Window", "window")}}.
```

File: files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+
+The frequency of calls to the callback function will generally match the display refresh rate. The most common refresh rate is 60hz, (60 cycles/frames per second), though 75hz, 120hz, and 144hz are also widely used. `requestAnimationFrame()` calls are paused in most browsers when running in background tabs or hidden {{HTMLElement("iframe")}}s, in order to improve performance and battery life.
+
+Your callback function must call `requestAnimationFrame()` again if you want to animate another frame. `requestAnimationFrame()` is one-shot.
+
+> **Warning:** Be sure always to use the first argument (or some other method for getting the current time) to calculate how much the animation will progress in a frame — **otherwise, the animation will run faster on high refresh-rate screens**. For ways to do that, see the examples below.
+
+Calling `requestAnimationFrame()` method requires that the current worker need to support this method, which means that the current worker is created by {{domxref("Window", "window")}} or by a dedicated worker that also supports this method.
```
Comment (skyclouds2001) [inline files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md:19]:
```suggestion
Calling `requestAnimationFrame()` method requires that the current worker need to has an associated owner {{domxref("Window", "window")}}, which means that the current worker must be created by {{domxref("Window", "window")}} or by a dedicated worker that also has an associated owner {{domxref("Window", "window")}}.
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
A few last grammar and typo fixes, and this is ready to be merged. Well done!

File: files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+A call to the `requestAnimationFrame()` method schedules only one single call to the callback function. If you want to animate another frame, your callback function must call `requestAnimationFrame()` again. 
+
+> **Warning:** Be sure always to use the first argument (or some other method for getting the current time) to calculate how much the animation will progress in a frame — **otherwise, the animation will run faster on high refresh-rate screens**. For ways to do that, see the examples below.
+
+Calling the `requestAnimationFrame()` method requires that the current worker need to has an associated owner {{domxref("Window", "window")}}, which means that the current worker must be created by {{domxref("Window", "window")}} or by a dedicated worker that also has an associated owner {{domxref("Window", "window")}}.
```
Comment (skyclouds2001) [inline files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md:19]:
```suggestion
Calling the `requestAnimationFrame()` method requires the current worker to have an associated owner {{domxref("Window", "window")}}. That means that the current worker must be created by {{domxref("Window", "window")}} or by a dedicated worker that also has an associated owner {{domxref("Window", "window")}}.
```

File: files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+  - The timestamp is a decimal number, in milliseconds, but with a minimal precision of 1 millisecond. The timestamp value is also similar to calling {{domxref('performance.now()')}} at the start of the callback function, but it is never the same value.
+  - When multiple callbacks queued by `requestAnimationFrame()` begin to fire in a single frame, each receives the same timestamp even though time has passed during the computation of every previous callback's workload.
+
+### Return value
+
+A `long` integer value that is the request ID uniquely identifying the entry
+in the callback list. This is a non-zero value, but you may not make any other
+assumptions about it. You can pass this value to
+{{domxref("DedicatedWorkerGlobalScope.cancelAnimationFrame()", "cancelAnimationFrame()")}} to cancel the refresh callback request, the cancel action must be made in the same worker.
```
Comment (skyclouds2001) [inline files/en-us/web/api/dedicatedworkerglobalscope/requestanimationframe/index.md:39]:
```suggestion
{{domxref("DedicatedWorkerGlobalScope.cancelAnimationFrame()", "cancelAnimationFrame()")}} to cancel the refresh callback request, the cancel action must have been made in the same worker.
```

**Review by teoli2003 — APPROVED** [review]
Thank you!

**Pull request merged**